primo/data_parser/well_data.py

```diff
@@ -864,12 +872,76 @@ def _process_input_data(self):
 
         LOGGER.info("Completed processing the essential inputs.")
 
+    def _get_state(self) -> str:
+        """
+        Attempts to infer the state for which the dataset belong.
+        Assumes that all data points belong to the same state and
+        we have at least one row of data
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        Two-digit code identifying the state for which the data belongs
+        """
+        wcn = self._col_names
+        lat = self.data[wcn.latitude].iloc[0]
+        long = self.data[wcn.longitude].iloc[0]
+        fips_code = get_fips_code(lat, long)
+        state = get_fips_part(fips_code, "STATE")
+        return state
+
+    def _convert_data_to_geodataframe(self) -> gpd.GeoDataFrame:
+        """
+        Returns a Geopandas GeoDataFrame object from well data
+        """
+        wcn = self._col_names
+        data = copy.deepcopy(self.data)
+        gdf = gpd.GeoDataFrame(
+            data,
+            geometry=gpd.points_from_xy(data[wcn.longitude], data[wcn.latitude]),
+            crs="EPSG:4326",
+        )
+        return gdf
+
     def _append_fed_dac_data(self):
         """Appends federal DAC data"""
-        census_year = self.config.census_year
-        # TODO:
-        # Append Tract ID/GEOID, population density, Total population, land area,
-        # federal DAC score to the DataFrame
+        if len(self.data) == 0:
+            # Nothing to do if no data
+            return
+
+        state_code = self._get_state()
+        census_tracts = get_state_census_tracts(state_code, DAC_TRACT_YEAR)
+        gdf = self._convert_data_to_geodataframe()
+
+        # Spatial join to identify tract id associated with every well
+        # Per 2010 data
+        gdf.sjoin(census_tracts, how="left", predicate="within")
```

Comment:
Suggesting registering new column names first:
```python
self._col_names.register_new_columns({
    "land_area": "Land Area [m2]",
    "water_area": "Water Area [units]",
    "geoid": "Census Tract ID [2010]",
})
self.data[self._col_names.geoid] = gdf["GEOID"]
```
Also, are all the columns needed for computing the DAC score? I don't see land area and water area being used here.

Reply:
They aren't being used anymore. I left them in mainly because the TODO included a suggestion to include them. Removing

Reply:
No, we don't have to register all the columns. We register columns only if we need to work with those columns later on. I assumed that `land_area` and `water_area` will be used later for calculations, so I suggested registering them.